db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/sqlgrammar.jj

```diff
@@ -10914,28 +10915,45 @@ tableReferenceTypes(boolean nestedInParens) throws StandardException :
 Object[]
 optionalTableClauses() throws StandardException :
 {
-    Object[]             otc = null;
-    Properties            tableProperties = null;
-    ResultColumnList    derivedRCL = null;
-    String                correlationName = null;
+    Object[]         otc = null;
+    Properties       tableProperties = null;
+    ResultColumnList derivedRCL = null;
+    String           correlationName = null;
+    ValueNode        txnIdExpression = null;
 }
 {
     otc = optionalTableProperties()
     {
         otc[OPTIONAL_TABLE_CLAUSES_DERIVED_RCL] = derivedRCL;
         otc[OPTIONAL_TABLE_CLAUSES_CORRELATION_NAME] = correlationName;
+        otc[OPTIONAL_TABLE_CLAUSES_TXN_ID] = txnIdExpression;
         return otc;
     }
 |
+    LOOKAHEAD( { getToken(1).kind != AS || getToken(2).kind != OF })
```

Comment:
Is it possible to remove the look ahead here?

Reply:
Would you please check my comment [here](https://github.com/splicemachine/spliceengine/pull/3693#discussion_r450398057)? I am not sure I can remove it actually but maybe I missed something.